### DISCUSSION TITLE
autoSyncPersistentDataPath parameter support

### DISCUSSION THREAD

**Discussion by Bart-Verdonck** [jeffreylanters/react-unity-webgl#567] [Feature Requests]
There's been a fix for an issue with synchronizing the IndexedDB [From issue tracker](https://issuetracker.unity3d.com/issues/webgl-streamwriter-not-triggering-syncfs-when-writing-a-file-to-slash-idbfs):

Fixed in [2021.3.44f1](https://unity.com/releases/editor/whats-new/2021.3.44#notes), [2022.3.44f1](https://unity.com/releases/editor/whats-new/2022.3.44#notes), [6000.0.11f1](https://unity.com/releases/editor/whats-new/6000.0.11#notes) (search for "IndexedDB" in release notes)

Fixed an issue that Application.persistentDataPath would not automatically persist, by adding a new JS config option `autoSyncPersistentDataPath: true` to enable automatic synchronization of Application.persistentDataPath over to IndexedDB.

This option is found in the Builds Index.html file, uncomment it and refresh the running build page and the issue will be resolved.

To use this new setting, you can modify the web templates provided by Unity for each installation you want this behavior in. Or better yet, make it team-shared by [adding a custom web template](https://docs.unity3d.com/6000.0/Documentation/Manual/webgl-templates.html) in Assets/WebGLTemplates in your project. Whatever you choose, add autoSyncPersistentDataPath: true, into the config that's passed into createUnityInstance() in the template's index.html.

The description for the new config [from the docs](https://docs.unity3d.com/6000.0/Documentation/Manual/webgl-templates.html#build_configuration):

`autoSyncPersistentDataPath: true`

If set to true, all file writes inside the Unity Application.persistentDataPath directory automatically persist so that the contents are remembered when the user revisits the site the next time. If unset (or set to false), you must manually sync file modifications inside the Application.persistentDataPath directory by calling the JS_FileSystem_Sync() JavaScript function.

It's unset (disabled) by default to not break existing projects. See `<Unity Install Location>\Editor\Data\PlaybackEngines\WebGLSupport\BuildTools\WebGLTemplates\Default\index.html` for the explanation:

// config.autoSyncPersistentDataPath = true;
This autosyncing is currently not the default behavior to avoid regressing existing user projects that might rely on the earlier manual JS_FileSystem_Sync() behavior, but in future Unity version, this will be expected to change.

**Comment by jeffreylanters** [comment]
A release with included support will be published next week. Thanks for your suggestion!

**Comment by jeffreylanters** [comment]
Hi! This feature has been released in update! [v9.7.0](https://github.com/jeffreylanters/react-unity-webgl/releases/tag/v9.7.0) including [updated docs](https://react-unity-webgl.dev/docs/api/auto-sync-persistent-data-path).
